core/include/detray/tools/navigator.hpp

```diff
@@ -533,35 +523,11 @@ class navigator {
     }
 
     DETRAY_HOST_DEVICE
-    auto &get_detector() { return detector; }
-
-    DETRAY_HOST_DEVICE
-    const auto &get_detector() const { return detector; }
+    const detector_t &get_detector() { return *_detector; }
```

Comment:
Shoot... I still gave a bad recipe. 😦  This should really rather be:

```c++
    DETRAY_HOST_DEVICE
    const detector_t &get_detector() const { return *_detector; }
```

I.e. the function should be `const` as well. Sorry about that...

Reply:
OK!